### PR TITLE
Elaborate on details of `HyperbandPruner`.

### PR DESCRIPTION
**Pull request by crcrpar**
This is a tiny PR that improves the documentation by adding some reference URLs.

### CODE REVIEW

File: optuna/pruners/hyperband.py [hunk 1]
```diff
@@ -24,6 +24,9 @@ class HyperbandPruner(BasePruner):
     Note that this implementation does not take as inputs the maximum amount of resource to
     a single SHA noted as :math:`R` in the paper.
 
+    This implementation can work with :class:`~optuna.samplers.TPESampler` and this combination
```
Comment (sile) [inline optuna/pruners/hyperband.py]: How about modifying a bit as follows (for users that don't know what `TPESampler` is)?
```suggestion
    This implementation can work with :class:`~optuna.samplers.TPESampler`, the default sampler, and this combination
```

File: optuna/pruners/hyperband.py [hunk 2]
```diff
@@ -50,6 +53,8 @@ class HyperbandPruner(BasePruner):
             :class:`~optuna.pruners.SuccessiveHalvingPruner`.
         n_brackets:
             The number of :class:`~optuna.pruners.SuccessiveHalvingPruner`\\ s (brackets).
+            For those who are curious why this defaults to :math`4`, see
```
Comment (hvy) [inline optuna/pruners/hyperband.py]: A nit, believing that we can afford to be more stiff and short here.
```suggestion
            Defaults to :math`4`. See
```

**Comment by crcrpar** [comment]
Thank you for your review. I added your suggestions.

File: optuna/pruners/hyperband.py [hunk 3]
```diff
@@ -24,6 +24,10 @@ class HyperbandPruner(BasePruner):
     Note that this implementation does not take as inputs the maximum amount of resource to
     a single SHA noted as :math:`R` in the paper.
 
+    This implementation works with :class:`~optuna.samplers.TPESampler`, the default sampler
+    and this combination seems to work well.
+    See: https://github.com/optuna/optuna/pull/828#issuecomment-575457360
```
Comment (crcrpar) [inline optuna/pruners/hyperband.py]: **Discussion**

This comment seems like aiming at those who are interested in Optuna or hyperparameter optimization. Therefore, this comment might confuse users who want to use Optuna as a hyperparameter tuning tool.

So, the content should be changed as follows:
- In Hyperband paper, the counterpart of RandomSampler is used.
- OTOH, Optuna uses TPE Sampler by default.
- The benchmark result shows that this implementation works with both Random Sampler and TPE Sampler.

Finally, this information might be better written in `note` block.

Comment (crcrpar) [reply optuna/pruners/hyperband.py]: Here's how the current documentation looks like.
image

File: optuna/pruners/hyperband.py [hunk 4]
```diff
@@ -24,6 +24,13 @@ class HyperbandPruner(BasePruner):
     Note that this implementation does not take as inputs the maximum amount of resource to
     a single SHA noted as :math:`R` in the paper.
 
+    .. note::
+        * In Hyperband paper, the counterpart of :class:`~optuna.samplers.RandomSampler` is used.
```
Comment (hvy) [inline optuna/pruners/hyperband.py]:
```suggestion
        * In the Hyperband paper, the counterpart of :class:`~optuna.samplers.RandomSampler` is used.
```

**Pull request merged**